rust/src/cli/service.rs

```diff
@@ -74,26 +75,35 @@ pub(crate) fn ncl_service(
     Ok("".to_string())
 }
 
-// All file ending with `.yml` will be included.
+// All file ending with `.yml` that do not have a symlink ending with `.applied`
+// will be included.
 fn get_config_files(folder: &str) -> Result<Vec<PathBuf>, CliError> {
     let folder = Path::new(folder);
-    let mut ret = Vec::new();
+    let mut yml_files = HashSet::<PathBuf>::new();
+    let mut applied_files = HashSet::<PathBuf>::new();
     for entry in folder.read_dir()? {
         let file = entry?.path();
         if file.extension() == Some(OsStr::new(CONFIG_FILE_EXTENTION)) {
-            ret.push(folder.join(file));
+            yml_files.insert(folder.join(file).with_extension(""));
+        } else if file.extension() == Some(OsStr::new(SYMLINK_FILE_EXTENTION)) {
+            applied_files.insert(folder.join(file).with_extension(""));
```

Comment:
I don't fully understand the implications of this change, but it's possible there is a subtle change in behavior here. Do we need to consider the case where a .yml file is deleted/recreated?

For example, previously if a .yml file was created, it moved to .applied after being acted on. What would happen in that case if a config of the same name was created again? or deleted then created? It seems like maybe we'd have a `.yml` file again which gets relocated to `.applied`.

do we need to consider this?

With the new behavior, if the .yml file is deleted then re-created, the presence of that symlink would prevent the .yml from being acted upon.

Reply:
Maybe we can store a hash(sha256sum) to `.applied` file for this?
@qinqon any idea?

Reply:
@davidvossel @cathay4t hash is a good idea, but we should also normalize the yaml before hashing it, I will update the PR.

Reply:
Updated PR, let's just not canonalize the yaml for now.

Reply:
@davidvossel Is the new code looks ok to you now?